molecule/cloud-aws-direct/playbook.yml

```diff
@@ -3,44 +3,173 @@
   hosts: all
   roles:
     - role: ansible-role-jenkins
-      jenkins2_pipeline_libraries_enabled: true
-      jenkins2_pipeline_libraries_url: 'https://example.com/example.git'
-      jenkins2_bitbucket_project_enabled: true
-      jenkins2_updates_url: http://ftp-nyc.osuosl.org/pub/jenkins/updates/
-      jenkins2_sonarqube_enabled: false
-      jenkins2_custom_files_enabled: true
-      jenkins2_gitlab_enabled: true
-      jenkins2_github_enabled: true
-      jenkins2_seed_job_enable: false
-      gitlab_external_url: 'https://localhost'
-      gitlab_project_group: 'test'
-      gitlab_project_name: 'test'
-      ci_test_jenkins_slave_label: 'ci_slaves'
-      feature_verification_jenkins_slave_label: 'fv_slaves'
-      ami_for_ci: 'ami-AAAAAAAA'
-      ami_for_fv: 'ami-BBBBBBBB'
-      ami_for_env: 'ami-CCCCCCCC'
-      ami_for_solr: 'ami-9887c6e7'
-      lb_security_groups: '"sg-AAAAAAAA","sg-CCCCCCCC"'
-      ec2_security_groups: '"sg-BBBBBBBB","sg-DDDDDDDD"'
-      aws_key: 'aws_key_name'
-      aws_subnet_id: 'subnet-AAAAAAAA'
-      artifact_bucket_name: 's3_artifact_test'
-      jenkins2_ec2_enable: true
-      jenkins2_jira_enabled: true
-      jenkins2_globaltools_enable: true
-      jenkins2_globaltools_nodejs_enable: false
-      jenkins2_globaltools_jdk_version:
-        - jdk-9.0.4-oth-JPR
-      jenkins2_globaltools_maven_version:
-        - 3.6.1
-      jenkins2_globaltools_ant_version:
-        - 1.10.6
-      jenkins2_globaltools_sonarscanner_version:
-        - 3.3.0.1492
-      jenkins2_globaltools_sonarscannermsbuild_version:
-        - 4.6.2.2108
-      jenkins2_globaltools_gradle_version:
-        - 5.5-rc-2
-      jenkins2_globaltools_docker_version:
-        - latest
+      jenkins2_configure_by_casc: true
+      jenkins2_configure_by_groovy: false
+      jenkins2_jcasc_config:
+        credentials:
+          system:
+            domainCredentials:
+            - credentials:
+              - basicSSHUserPrivateKey:
+                  description: "credentials for ubuntu user to connect remotely"
+                  id: "sshconnection"
+                  privateKeySource:
+                    directEntry:
+                      privateKey: "-----BEGIN RSA PRIVATE KEY-----
```

Comment:
please give explanation for this private key

Reply:
This key was placed here as an example, nevertheless,  I have changed all entries to 
"-----BEGIN RSA PRIVATE KEY-----
Here could be placed your private key,
but we strongly recommend keeping private keys in vault service,
like Ansible-vault or Hashicorp-vault, e.t.c
-----END RSA PRIVATE KEY-----"
in the last commit.